TF-Seq2Seq via Docker

ReDeiPirati opened the pull request:
I have provided an alternative method of installation via Docker(documented in the docs); i've tested all the Dockerfile image. This pull contains also the #254 otherwise the python3.5 or a Tensorflow 1.2 version would fail the test.

darrengarvey committed:
Fix typo in hooks_test.py.

darrengarvey committed:
Update hooks_test.py to work with TensorFlow 1.2.

darrengarvey committed:
Update imports to support TensorFlow v1.2.

darrengarvey committed:
Fix hooks_test.py on Python 3.

Fixes the following error when running the tests on Python 3.

    Traceback (most recent call last):
      File "seq2seq/seq2seq/test/hooks_test.py", line
      55, in test_begin
          self.assertEqual(file_contents.decode(), ...)
          AttributeError: 'str' object has no attribute 'decode'

darrengarvey committed:
Another go at fixing hooks_test.py.

distutils.version isn't available on the CI versions, so change the test
to not rely explicitly on the version of TensorFlow installed.

ReDeiPirati committed:
Now is possible to run Tf-Seq2Seq via Docker

ReDeiPirati committed:
Fix typo on Docker README

ReDeiPirati committed:
Add PackageName in README and fix typo in nmt

ReDeiPirati committed:
Fix some typo and Update the Getting started with the Docker installation

ReDeiPirati committed:
Fix Dockefile missing /n

ReDeiPirati commented:
I signed it!

ReDeiPirati commented:
Well this is strange because i haven't changed a single line of python code and during test i've never failed a pylint test. But now it fails even in the Docker container as during CI pipeline. @darrengarvey may you help me? 

Edit: I've opened a [Issue](https://github.com/PyCQA/pylint/issues/1542) on pylint beacuse both the errors seem to be False positive.

darrengarvey committed:
Tell pylint about the tf contextmanager.

Use of `tf.name_scope` and `tf.variable_scope` cause pylint errors on TF1.2
due to pylint not fully understanding the `tf_contextlib.contextmanager`
decorators.

darrengarvey commented:
@ReDeiPirati - Looks like it's started failing because the tensorflow pip package was updated to v1.2.0 on the 15th. My PR passed Circle CI checks because it was still building against TF 1.1 (while I had TF 1.2 locally).

I'll take a look.

darrengarvey committed:
Work around no-name-in-module pylint errors.

Following some changes in TF to the `LazyLoader` [1], pylint complains
about not being able to find some imports under `tf.contrib`. This looks
like a pylint issue, emitting errors like:

    ************* Module seq2seq.encoders.rnn_encoder
    E: 24, 0: No name 'rnn' in module 'LazyLoader' (no-name-in-module)
    ************* Module seq2seq.data.input_pipeline
    E: 32, 0: No name 'slim' in module 'LazyLoader' (no-name-in-module)

[1] https://github.com/tensorflow/tensorflow/commit/95c5d7e880b8b4d18ba1f3b7cf40d15cd218b3c9

darrengarvey commented:
@ReDeiPirati - I pushed a couple of fixes to #254. The `no-name-in-module` errors I just had to work around though. The LazyLoader is possibly too dynamic for pylint now, eg:

```python
# OK:
from tensorflow.contrib import learn
learn.MetricSpec
# pylint errors with: "No name 'learn' in module 'LazyLoader' (no-name-in-module)"
from tensorflow.contrib.learn import MetricSpec
```

ReDeiPirati committed:
Merge branch 'TF1.2-fixes' of https://github.com/darrengarvey/seq2seq into Add-Docker

ReDeiPirati committed:
Fix typo

ReDeiPirati commented:
@darrengarvey Thank you for your help!! Now it works fine.
